File: src/main/angular/src/app/address/model/address.ts
import {City, CityImpl} from './city';

export interface Address {
  id: number;
  name: string;
  street: string;
  number: string;
  city: City;
}

export class AddressImpl implements Address {
  id: number;
  name: string;
  street: string;
  number: string;
  city: City;

  constructor(name: string, street: string, number: string, zip: string, city: string) {
    this.name = name;
    this.street = street;
    this.number = number;
    this.city = new CityImpl(zip, city);
  }
}

File: src/main/java/be/tvl/address/Address.java
package be.tvl.address;

import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    private String name;

    private String street;

    private String number;

    @Embedded
    private City city;

    public Address() {
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getStreet() {
        return street;
    }

    public String getNumber() {
        return number;
    }

    public City getCity() {
        return city;
    }
}
